.github/workflows/checkAndSubmitAddonMetadata.yml

```diff
@@ -179,8 +179,179 @@ jobs:
+          Reviews should abide by the [NVDA code of conduct](https://github.com/nvaccess/nvda/blob/master/CODE_OF_CONDUCT.md).
+    - name: Add add-on ID and discussion data to json file
+      if: failure() && steps.checkDiscussion.outcome == 'failure'
+      run: |
+        addonId=$(
+          echo ${{ steps.getAddonNameAndVersion.outputs.addonName }} \
+         )
+        discussionId=$(
+          echo ${{ steps.createDiscussion.outputs.discussion-id }}
+        )
+        discussionUrl=$(
+          echo ${{ steps.createDiscussion.outputs.discussion-url }}
+        )
+        jqCode="
+        .[\"$addonId\"].discussionId = \"$discussionId\"
+        | .[\"$addonId\"].discussionUrl = \"$discussionUrl\"
+        "
+  
+        mv discussions.json discussions.old.json
+        jq -e "$jqCode" discussions.old.json > discussions.json
+        jqExitCode=$?
+        rm discussions.old.json
+        exit $jqExitCode
+    - name: Create review comment
+      id: createComment
+      if: always()
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const fs = require('fs')
+          const addonVersion = "${{ steps.getAddonNameAndVersion.outputs.addonVersion }}"
+          const contents = fs.readFileSync('discussions.json')
+          const data = JSON.parse(contents)
+          const addonName = "${{ steps.getAddonNameAndVersion.outputs.addonName }}"
+          const discussionId = data[addonName]["discussionId"]
+          console.log(`${discussionId}`)
+          const mutation = `mutation {
+            addDiscussionComment(
+              input: {
+                body: "Reviews for version ${addonVersion}",
+                discussionId: "${discussionId}",
+                clientMutationId: "GitHub Script action"
+              }
+            ) {
+              clientMutationId
+              comment {
+                url
+              }
+            }
+            }`;
+          const response = await github.graphql(mutation)
+          const url = response.addDiscussionComment.comment.url
+          core.setOutput('url', url)
+    - name: Add discussion URL to metadata
+      if: always()
+      run: |
+        addonFilename=$(
+          echo ${{ needs.getAddonFilename.outputs.addonFileName }}
+        )
+        reviewUrl=$(
+          echo ${{ steps.createComment.outputs.url }}
+        )
+        jqCode="
+        .[\"reviewUrl\"] = \"$reviewUrl\"
+        "
+  
+        mv $addonFilename $addonFilename.old.json
+        jq -e "$jqCode" $addonFilename.old.json > $addonFilename
+        jqExitCode=$?
+        rm $addonFilename.old.json
+        exit $jqExitCode
+    - name: Commit and push
+      if: always()
+      env:
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        git config user.name github-actions
+        git config user.email github-actions@github.com
+        git add .
+        git commit -m "update discussion URL"
+        git checkout -b updateDiscussionUrl
```

Comment:
this seems unnecessary / is replaced by the `gh pr checkout`

```suggestion
```

Reply:
We need to use a branch, not a commit in detached state. Otherwise, when we perform checkout, the last commit cannot be merged. So I have put directly the branch corresponding to the PR as ref of checkout action.